src/components/autocompleter/autocompleter.test.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { createRef } from 'react';
+import {
+  render, screen,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import jestServer from '../../mocks/jest-server.js';
+import Autocompleter from './autocompleter.jsx';
+import mockHandlers from './mock-handlers.js';
+
+describe('src/components/autocompleter/autocompleter', () => {
+  beforeEach(() => {
+    jestServer.use(...mockHandlers());
+  });
+
+  const requiredProps = {
+    id: 'test-id',
+    label: 'Test label',
+    name: 'field-id',
+    apiEndpoint: '/models',
+  };
+
+  it('has defaults', () => {
+    const ref = createRef();
+    render(<Autocompleter {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('className API', () => {
+    it('prioritizes className prop', () => {
+      const { container } = render(<Autocompleter {...requiredProps} className="prioritize-me" />);
+      expect(container.firstChild).toHaveClass('prioritize-me');
+    });
+  });
+
+  describe('errorText', () => {
+    it('sets the input to be invalid', () => {
+      render(<Autocompleter {...requiredProps} errorText="not valid" />);
+      expect(screen.getByLabelText('Test label')).not.toBeValid();
+    });
+  });
+
+  describe('id API', () => {
+    it('accepts an ID', () => {
+      render(<Autocompleter {...requiredProps} id="this-is-an-id" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('id', 'this-is-an-id');
+    });
+  });
+
+  describe('label API', () => {
+    it('accepts a label', () => {
+      render(<Autocompleter {...requiredProps} label="Bar" />);
+      expect(screen.getByLabelText('Bar')).toBeDefined();
```

Comment:
info for myself - is there a reason to use `toBeDefined` over `toBeInTheDocument`?

It's sixes in this test since the `get` would fail, but I've seen this elsewhere and been curious about it

Reply:
I think we prefer `toBeInTheDocument` here just for the sake of affordance. But @juanca and @Bloodyaugust probably know better than I.

Reply:
Yeah, `toBeInTheDocument()` is preferred. 

Technically, `toBeInTheDocument()` matcher checks whether it's actually in the document. This can be useful when you find an element, do some user interaction, and test that it is _still_ in the document or maybe that it was yeeted.

Reply:
But maybe we can improve this test. We really want to assert that the label has a control.

So maybe:

```suggestion
      expect(screen.getByText('Bar').control).toBe(document.getById('test-id'));
```